### PR TITLE
sync with latest xarray

### PR DESCRIPTION
**Pull request by malmans2**
`sarsen` is not compatible with the latest (unreleased) `xarray`.
I was able to report and fix a couple of issues upstream (https://github.com/pydata/xarray/issues/6597, https://github.com/pydata/xarray/issues/6600), but we need to apply a few changes in `sarsen` as well.

I tested with `xarray@main` commit `c34ef8a60227720724e90aa11a6266c0026a812a`, and all tests are successful.

Let me know if you would like me to add an action that (periodically?) runs the CI using the `xarray` version under development.

### CODE REVIEW

File: sarsen/radiometry.py [hunk 1]
```diff
@@ -18,14 +18,13 @@ def sum_weights(
     multilook: T.Optional[T.Tuple[int, int]] = None,
 ) -> xr.DataArray:
     geocoded = initial_weights.assign_coords(
-        slant_range_index=slant_range_index, azimuth_index=azimuth_index
+        # Use z_level names for backward compatibility with xarray<=2022.03.0
+        z_level_0=slant_range_index,
+        z_level_1=azimuth_index,
```
Comment (malmans2) [inline sarsen/radiometry.py:23]: I'm pretty sure this is needed when the `multilook` option is turned on. `xarray<=2022.03.0` and therefore `sarsen` uses fixed names ("z_level_0", "z_level_1"), whereas the latest version of xarray would use ("slant_range_index", "azimuth_index").

I don't think `multilook` is tested by the CI though, so I haven't really tested this.

**Comment by malmans2** [comment]
As discussed with @aurghs, the code is now simpler but we'll have to ping `xarray > 2022.03.0`.

TODO: Fix `mypy`, which is currently raising a few issues. Many typings have been introduced/improved in the latest `xarray`

**Comment by malmans2** [comment]
The other mypy issue is being fixed upstream: https://github.com/pydata/xarray/issues/6628

File: sarsen/radiometry.py [hunk 2]
```diff
@@ -172,7 +174,7 @@ def gamma_weights_nearest(
 
 
 def azimuth_slant_range_grid(
-    measurement_ds: xr.Dataset,
+    measurement_ds: xr.DataArray,
```
Comment (malmans2) [inline sarsen/radiometry.py:177]: @alexamici looks like this should be a DataArray but the variable name is confusing

Comment (alexamici) [reply sarsen/radiometry.py:177]: D'oh! I'll change it outside of this PR

**Pull request merged**